src/Microsoft.Extensions.Configuration.AzureAppConfiguration/AzureAppConfigurationProvider.cs

```diff
@@ -883,5 +897,42 @@ private bool IsFailOverable(RequestFailedException rfe)
                    rfe.Status == (int)HttpStatusCode.RequestTimeout ||
                    rfe.Status >= (int)HttpStatusCode.InternalServerError;
         }
+
+        private async Task<Dictionary<string, ConfigurationSetting>> MapConfigurationData(Dictionary<string, ConfigurationSetting> data)
+        {
+            Dictionary<string, ConfigurationSetting> mappedData = new Dictionary<string, ConfigurationSetting>(StringComparer.OrdinalIgnoreCase);
+
+            foreach (KeyValuePair<string, ConfigurationSetting> kvp in data)
+            {
+                ConfigurationSetting setting = kvp.Value;
+                foreach (Func<ConfigurationSetting, ValueTask<ConfigurationSetting>> func in _options.UserDefinedMappers)
+                {
+                    setting = await func(setting).ConfigureAwait(false);
+                }
+                mappedData[kvp.Key] = setting;
+            }
+
+            return mappedData;
+        }
+
+        private void UpdateWatchedSettings()
+        {
+            _watchedSettings.Clear();
+
+            foreach (KeyValueWatcher changeWatcher in _options.ChangeWatchers)
+            {
+                string watchedKey = changeWatcher.Key;
+                string watchedLabel = changeWatcher.Label;
+                KeyValueIdentifier watchedKeyLabel = new KeyValueIdentifier(watchedKey, watchedLabel);
+
+                ConfigurationSetting watchedKvCopy = _serverData.Values.FirstOrDefault((kv) => kv.Key.Equals(watchedKey) && kv.Label == watchedLabel);

```

Comment:
>watchedKvCopy [](http://example.com/codeflow?start=37&length=13)

Nit: watchedKv 

You can add a comment that the watchedKv will be a deep copy of the original setting from server because we are extracting from _serverData dictionary.